lib/cache/collections.go

```diff
@@ -1740,3 +1745,73 @@ func (c *authPreference) processEvent(ctx context.Context, event services.Event)
 func (c *authPreference) watchKind() services.WatchKind {
 	return c.watch
 }
+
+type clusterNetworkingConfig struct {
+	*Cache
+	watch services.WatchKind
+}
+
+func (c *clusterNetworkingConfig) erase(ctx context.Context) error {
+	if err := c.clusterConfigCache.DeleteClusterNetworkingConfig(ctx); err != nil {
+		if !trace.IsNotFound(err) {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+
+func (c *clusterNetworkingConfig) fetch(ctx context.Context) (apply func(ctx context.Context) error, err error) {
+	var noConfig bool
+	resource, err := c.ClusterConfig.GetClusterNetworkingConfig(ctx)
+	if err != nil {
+		if !trace.IsNotFound(err) {
+			return nil, trace.Wrap(err)
+		}
+		noConfig = true
+	}
+	return func(ctx context.Context) error {
+		// either zero or one instance exists, so we either erase or
+		// update, but not both.
+		if noConfig {
+			if err := c.erase(ctx); err != nil {
+				return trace.Wrap(err)
+			}
+			return nil
+		}
+
+		c.setTTL(resource)
+		if err := c.clusterConfigCache.SetClusterNetworkingConfig(ctx, resource); err != nil {
+			return trace.Wrap(err)
+		}
+		return nil
+	}, nil
+}
+
+func (c *clusterNetworkingConfig) processEvent(ctx context.Context, event services.Event) error {
+	switch event.Type {
+	case backend.OpDelete:
+		err := c.clusterConfigCache.DeleteClusterNetworkingConfig(ctx)
+		if err != nil {
+			if !trace.IsNotFound(err) {
+				c.Warningf("Failed to delete resource %v.", err)
```

Comment:
why is this the only error logged?
consider removing the log, caller should be doing that

Reply:
This was copied over from another resource. Since these functions are almost identical among the resources, I'd suppose it's preferable to keep the cache handlers consistent/uniform.